typescript/src/pages/SharedPage.tsx

```diff
@@ -0,0 +1,44 @@
+import { useGetFolder } from "folder/data-access-folder";
+import { Layout } from "sharing/feature-layout";
+import { SharedLayout } from "page-layout/SharedLayout";
+import { CardList } from "link/ui-card-list";
+import { FolderInfo } from "folder/ui-folder-info";
+import { ReadOnlyCard } from "link/ui-read-only-card";
+import { SearchBar } from "link/ui-search-bar";
+import { ReactNode } from "react";
+
+interface link {
+  id: number;
+  createdAt: string;
+  url: string;
+  title: string;
+  description: string;
+  imageSource?: ReactNode;
+}
+
+export const SharedPage = () => {
+  const { data }: { data: any } = useGetFolder();
+  const { profileImage, ownerName, folderName, links } = data || {};
+
+  return (
+    <Layout>
+      <SharedLayout
+        folderInfo={
+          <FolderInfo
+            profileImage={profileImage}
+            ownerName={ownerName}
+            folderName={folderName}
+          />
+        }
+        searchBar={<SearchBar />}
+        cardList={
+          <CardList>
+            {links?.map((link: link) => (
+              <ReadOnlyCard key={link?.id} {...link} />
```

Comment:
links라는 데이터의 타입이 제대로 잡혀있다면, map의 콜백함수에 타입을 적어주지 않아도 되는데요~ 이런식으로 쓸 수 있게 코드를 작성해야됩니다!
```suggestion
            {links?.map((link) => (
              <ReadOnlyCard key={link?.id} {...link} />
```